cfme/tests/infrastructure/test_config_management_rest.py

```diff
@@ -230,3 +231,69 @@ def test_authentications_options(self, appliance, config_manager):
         collection = appliance.rest_api.collections.authentications
         assert 'credential_types' in collection.options()['data']
         assert_response(appliance)
+
+
+@pytest.fixture(scope="module")
+def config_manager_rest(config_manager_obj):
+    """Creates provider using REST API."""
+    config_manager_obj.create_rest()
+    assert_response(config_manager_obj.appliance)
+    rest_entity = config_manager_obj.rest_api_entity
+    rest_entity.reload()
+
+    yield rest_entity
+
+    if rest_entity.exists:
+        rest_entity.action.delete()
+        rest_entity.wait_not_exists()
```

Comment:
optional:
but not a good idea to wait for delete in tear-down... it if fails then actual test will fail.

Reply:
On 5.10, tower takes a little time to get deleted, due to which the one of the test fails. Need to use `wait_not_exists`.